mozilla_addon_signer/bugzilla.py

```diff
@@ -4,41 +4,82 @@
 class BugzillaAPI(object):
     api_base = 'https://bugzilla.mozilla.org/rest'
 
+    class APIException(Exception):
+        pass
+
     def __init__(self, api_key=None):
         self.session = requests.Session()
         if api_key:
             self.session.headers.update({'X-BUGZILLA-API-KEY': api_key})
 
-    def get(self, url, params=None):
-        res = self.session.get(self.api_base + url, params=params)
+    def request(self, method, endpoint, params=None, data=None, json=None):
+        url = self.api_base + endpoint
+        res = self.session.request(method, url, params=params, json=json, data=data)
         res.raise_for_status()
-        return res.json()
+        data = res.json()
+        if data.get('error', False):
+            raise self.APIException(data)
+        return data
 
-    def post(self, url, data=None):
-        res = self.session.post(self.api_base + url, data=data)
-        res.raise_for_status()
-        return res.json()
+    def get(self, endpoint, params=None):
+        return self.request('GET', endpoint, params=params)
+
+    def post(self, endpoint, json=None, data=None):
+        return self.request('POST', endpoint, json=json, data=data)
```

Comment:
It's subtle, but this causes most data to be JSON encoded instead of form encoded. Form encoding works for a lot of things, but according to bugzilla developers, JSON encoding is preferred for most things, and is required for things like setting flags.

The only thing that stays form encoded with this patch is uploading attachments. They may also be able to be converted to JSON, but it would have been harder and I didn't want to deal with that, since it worked fine form encoded.